In [1]:
# import the modules
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.metrics import confusion_matrix, classification_report

In [3]:
# read the csv file from the resources folder into a pandas DataFrame
lending_data_df = pd.read_csv(("lending_data.csv"))

# review the DataFrame
lending_data_df.head()

loan_size  interest_rate  borrower_income  ...  derogatory_marks  total_debt  loan_status
0    10700.0          7.672            52800  ...                 1       22800            0
1     8400.0          6.692            43600  ...                 0       13600            0
2     9000.0          6.963            46100  ...                 0       16100            0
3    10700.0          7.664            52700  ...                 1       22700            0
4    10800.0          7.698            53000  ...                 1       23000            0

[5 rows x 8 columns]

In [4]:
# seperate the data into labels and features

# seperate the y variable, the labels
y = lending_data_df['loan_status']

# seperate the X variable, the features
X = lending_data_df.drop(columns="loan_status")

In [5]:
# import the train_test_learn module
from sklearn.model_selection import train_test_split

# split the data using train_test_split
# assign a random_state of 1 to the function
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=1)

In [6]:
# import the LogisticRegression module from SKlearn
from sklearn.linear_model import LogisticRegression

# instantiate the Logistic Regression model
# assign a random_state parameter of 1 to the model
lr_model = LogisticRegression(random_state=1)

# fit the model using training data
lr_model.fit(X_train, y_train)

LogisticRegression(random_state=1)

In [7]:
# make a prediction using the testing data
y_pred = lr_model.predict(X_test)

In [8]:
# generate a confusion matrix for the model
confusion_matrix(y_test, y_pred)

array([[18655,   110],
       [   36,   583]])

In [10]:
# print the classification report for the model
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      0.99      1.00     18765
           1       0.84      0.94      0.89       619

    accuracy                           0.99     19384
   macro avg       0.92      0.97      0.94     19384
weighted avg       0.99      0.99      0.99     19384



Answer the following question: How well does the logistic regression model predict both the 0 (healthy loan) and 1 (high-risk loan) labels?

- The model excelled at predicting the healthy loans with 100% precision and 99% recall. The model was not as good at predicting high-risk loans at 84% precision and 94% recall.